src/devices/RGBLedMatrix/samples/RGBLedMatrix.sample.cs

```diff
@@ -0,0 +1,678 @@
+                            Vector3 cv = Star(uv, time);
+                            Color c = ToSRGB(cv.X, cv.Y, cv.Z);
+                            matrix.SetPixel(ix, iy, c.R, c.G, c.B);
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+        }
+
+        private static Vector3 clamp(Vector3 c, float a, float b)
+        {
+            return new Vector3(Math.Clamp(c.X, a, b), Math.Clamp(c.Y, a, b), Math.Clamp(c.Z, a, b));
+        }
+
+        private static float mod(float a, float b)
+        {
+            float ret = a % b;
+            if (ret < 0)
+            {
+                ret += b;
+            }
+
+            return ret;
+        }
+
+        private static Vector3 mod(Vector3 a, float b)
+        {
+            return new Vector3(mod(a.X, b), mod(a.Y, b), mod(a.Z, b));
+        }
+
+        private static Vector3 Add(Vector3 v, float s)
+        {
+            return new Vector3(v.X + s, v.Y + s, v.Z + s);
+        }
+
+        private static Vector3 abs(Vector3 vector)
+        {
+            return new Vector3(Math.Abs(vector.X), Math.Abs(vector.Y), Math.Abs(vector.Z));
+        }
+
+        private static Vector3 mix(Vector3 a, Vector3 b, float f)
+        {
+            return a * (1 - f) + b * f;
+        }
+
+        private static Vector3 hsv2rgb_smooth(Vector3 c)
+        {
+            float c1 = c.X + 6.0f;
+
+            Vector3 v1 = Add(new Vector3(0.0f, 4.0f, 2.0f), c.X * 6.0f);
+            Vector3 rgb = clamp(Add(abs(Add(mod(v1, 6.0f), -3.0f)), -1.0f), 0.0f, 1.0f);
+
+            rgb = rgb*rgb*(Add(-2.0f * rgb, 3.0f)); // cubic smoothing
+
+            return c.Z * mix(new Vector3(1.0f, 1.0f, 1.0f), rgb, c.Y);
+        }
+
+        private static float smoothstep(float edge0, float edge1, float x)
+        {
+            // Scale, bias and saturate x to 0..1 range
+            x = Math.Clamp((x - edge0) / (edge1 - edge0), 0.0f, 1.0f);
+            // Evaluate polynomial
+            return x * x * (3 - 2 * x);
+        }
+
+        private static Vector3 Star(Vector2 uv, float time)
+        {
+            Vector2 p = uv - new Vector2(0.5f, 0.5f);
+            float a = (float)(Math.Atan2(p.Y, p.X) / 2f / Math.PI);
+
+            a = mod(a + time / 10.0f, 1.0f);
+
+            float r = p.Length();
+            float n = 5.0f;
+
+            float s1 = 0.2f;
+            float s2 = 0.5f;
+            float srange = s2 - s1;
+            float x = Math.Abs(mod(a, 1.0f / n) * n - 0.5f) * srange / 0.5f + s1;
+
+            float blur = 4.0f + 3.0f * (float)Math.Sin(time * 2f * Math.PI / 5f);
+            float c = smoothstep(0.0f, 1.0f, (x - r) * blur);//x <= r ? 0.0 : 1.0;
+            //// return hsv2rgb_smooth(new Vector3(mod(time / 11.0f, 1.0f), 1.0f, c));
```

Comment:
commented out code

Reply:
I was leaving it for you specifically :-) but I'll remove all commented code.

Reply:
can't remember what this was doing anymore :smile:

Reply:
Cool. I removed the comments anyway as nobody will know what was this for :-)